server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/webhook/WebhookPostTask.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.task.projectanalysis.webhook;
+
+import com.google.common.collect.Iterables;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import org.sonar.api.ce.posttask.PostProjectAnalysisTask;
+import org.sonar.api.config.Settings;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.core.config.WebhookProperties;
+import org.sonar.core.util.stream.Collectors;
+import org.sonar.server.computation.task.projectanalysis.component.SettingsRepository;
+import org.sonar.server.computation.task.projectanalysis.component.TreeRootHolder;
+
+import static com.google.common.base.Throwables.getRootCause;
+import static java.lang.String.format;
+
+public class WebhookPostTask implements PostProjectAnalysisTask {
+
+  private static final Logger LOGGER = Loggers.get(WebhookPostTask.class);
+
+  private final TreeRootHolder rootHolder;
+  private final SettingsRepository settingsRepository;
+  private final WebhookCaller caller;
+
+  public WebhookPostTask(TreeRootHolder rootHolder, SettingsRepository settingsRepository, WebhookCaller caller) {
+    this.rootHolder = rootHolder;
+    this.settingsRepository = settingsRepository;
+    this.caller = caller;
+  }
+
+  @Override
+  public void finished(ProjectAnalysis analysis) {
+    Settings settings = settingsRepository.getSettings(rootHolder.getRoot());
+
+    Iterable<String> webhookProps = Iterables.concat(
+      getWebhookProperties(settings, WebhookProperties.GLOBAL_KEY),
+      getWebhookProperties(settings, WebhookProperties.PROJECT_KEY)
+    );
+    if (!Iterables.isEmpty(webhookProps)) {
+      process(settings, analysis, webhookProps);
+    }
+  }
+
+  private static List<String> getWebhookProperties(Settings settings, String propertyKey) {
+    String[] webhookIds = settings.getStringArray(propertyKey);
+    return Arrays.stream(webhookIds)
+      .map(webhookId -> format("%s.%s", propertyKey, webhookId))
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "%s.%s" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1192)